src/wp-includes/plugin.php

```diff
@@ -21,8 +21,7 @@
  * @since 1.5.0
  */
 
-// Initialize the filter globals.
-require __DIR__ . '/class-wp-hook.php';
+require_once ABSPATH . 'wp-includes/class-wp-autoload.php';
```

Comment:
This is curious -- is `src/wp-includes/plugin.php` included/required before the wp-autoload is set? And that's why it is necessary to include it here as well?

Reply:
I don't remember exactly the scenario that required this... But in all cases where I added this line in various files, it was because without it I was getting errors either when using WP, or when running PHPUnit tests 🤔 
I don't remember exactly the case, but this is one of the files that plugins/themes frequently include manually in order to load various functions, so maybe that's why I was getting errors 🤷

Reply:
> is `src/wp-includes/plugin.php` included/required before the wp-autoload is set? And that's why it is necessary to include it here as well?

It's not unknown for sites to include `plugin.php` in their `wp-config.php` file (eg [Altis default config](https://github.com/humanmade/altis-cms/blob/master/wp-config.php#L25)) so I think this is the result of an earlier suggestion.

It's probably worth a comment to the effect so it doesn't get removed unintentionally. A unit test to ensure it's in the file would be great too.

Reply:
Fair point. Comment added 👍